server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java

```diff
@@ -219,7 +219,7 @@ private Instant parseDateTime(String value, ZoneId timeZone, boolean roundUpIfNo
 
                 return DateFormatters.from(accessor).withZoneSameLocal(timeZone).toInstant();
             }
-        } catch (IllegalArgumentException | DateTimeParseException e) {
+        } catch (IllegalArgumentException | DateTimeException e) {
```

Comment:
In the [main parse method](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java#L185) we just catch `Exception`, and throw `IllegalArgumentException` - could we just do the same here?

Reply:
I'm not sure, catching `Exception` kinda makes me feel uneasy usually, is it ok in this context? Usually I prefer to specify.